### PR TITLE
Improved editor support for token files through JSON schemas generated via style-dictionary

### PR DESCRIPTION
**Pull request by jakobw**
Generating a JSON schema describing the structure of token files and including all token names makes it possible for text editors to provide autocompletion in token JSON files. This is especially useful for token references (e.g. `{font.line-height.style.label.value}`) which would otherwise be typed by hand.

Screenshot 2020-10-24 at 22 56 52

## Want!! How??
Letting the text editor know which JSON files to apply the schema to requires some configuration. I tried it out with VS Code and Intellij.

### VS Code
VS Code has good [documentation](https://code.visualstudio.com/docs/languages/json#_json-schemas-and-settings) on getting better editor support for JSON files and working with JSON schemas. TL;DR paste the following into your JSON workspace settings:
```json
{
    "json.schemas": [
        {
            "fileMatch": [
                "tokens/properties/**/*.json"
            ],
            "url": "./tokens/dist/schema.json"
        }
    ]
}
```

### Intellij/Webstorm/PHPStorm
Create an entry in your JSON schema settings.

Screenshot 2020-10-25 at 11 42 53

### CODE REVIEW

**Comment by micgro42** [comment]
That Idea looks nice! Your PR description should probably become some form of `.md` doc in the repo itself. Maybe a part of the readme? :thinking:

**Comment by jakobw** [comment]
@micgro42 thanks, I added the relevant parts to the tokens README!

File: tokens/README.md [hunk 1]
```diff
@@ -22,3 +22,29 @@ The tokens can be inspected by looking at the source files in the [`properties`
...
+                "tokens/properties/**/*.json"
+            ],
+            "url": "./tokens/dist/schema.json"
+        }
+    ]
+}
+```
+
+### JetBrains products
+Go to Settings and navigate to `Languages & Frameworks` > `Schemas and DTDs` >  `JSON Schema Mappings`. There you'll be able to create a mapping for the token files with the schema pointing to `./tokens/dist/schema.json` and the directory set to `tokens/properties/`.
```
Comment (itamargiv) [inline tokens/README.md:50]: If possible please move this to a different file other than the README (perhaps somewhere in the storybook docs) As this file is used as our package description on NPM and this information is not relevant to potential consumers of the token package.

**Review by itamargiv — CHANGES_REQUESTED** [review]
Woohoo \o/ This is brilliant. Thank you @jakobw for the initiative!!

The only comment below is about the location of the docs. otherwise, it's ready to merge.

File: tokens/README.md (hunk 1 above)
Comment (jakobw) [reply tokens/README.md]: Thanks, that's a good point! I moved it to "Developing Components" below the "Working with Component-level Design Tokens" section. I don't think that's a perfect fit, but it's the best place I found at the moment. I could imagine splitting the contents of this page up into multiple pages under a more generic "Development" headline at some point in the future.

**Review by itamargiv — APPROVED** [review]

**Comment by itamargiv** [comment]
Thank you! @jakobw

**Pull request merged**